### PR TITLE
Merge code for Kmer type from BioSequences to this package

### PR DESCRIPTION
**Pull request by BenJWard**

### CODE REVIEW

File: src/kmer.jl [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+###
+### Old Mer Base Functions - not transferred to new type.
+###
+#@inline encoded_data_type(::Type{Mer{A,K}}) where {A,K} = UInt64
+#@inline encoded_data_type(::Type{BigMer{A,K}}) where {A,K} = UInt128
+#@inline encoded_data_type(x::AbstractMer) = encoded_data_type(typeof(x))
+#@inline encoded_data(x::AbstractMer) = reinterpret(encoded_data_type(typeof(x)), x)
+#@inline ksize(::Type{T}) where {A,K,T<:AbstractMer{A,K}} = K
+#@inline Base.unsigned(x::AbstractMer) = encoded_data(x)
```
Comment (jakobnissen) [inline src/kmer.jl:271]: Some commented out sections here, and below, and also a bunch of whitespace above this.

File: src/transformations.jl [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+@inline function reverse_complement(x::Kmer{A,K,N}) where {A,K,N}
+    return complement(reverse(x))
+end
+
+#=
+@inline function reverse_complement2(seq::Kmer{A,K,N}) where {A,K,N}
+    f = x -> complement_bitpar(x, A())
+    rdata = _reverse(f, BitsPerSymbol(seq), seq.data...)
+    return Kmer{A,K,N}(rightshift_carry(rdata, 64N - 2K))
+end
```
Comment (jakobnissen) [inline src/transformations.jl:57]: Why two implementations?

**Review by jakobnissen — APPROVED** [review]
I've just given it a quick skim (too much code to deeply review) - it all looks great. I'm going to give it a test drive now :)

**Comment by jakobnissen** [comment]
I should get moving on that BioSequences update. It seems like you lean on the interface for this package!

**Comment by BenJWard** [comment]
I'd prefer if BioSequences hit v3 before I properly massage this to work with the new interface yeah. Then once this is a decent version based on v3, I can work on kmer-counting, spectras, and the genome assembly stuff again.

**Pull request merged**